### PR TITLE
Added BpkRating for Compose

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

**Comment by bvitaliyg** [comment]
Record snapshots

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/rating/BpkRating.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      overflow = TextOverflow.Ellipsis,
+    )
+  }
+}
+
+@Composable
+fun BpkRating(
+  value: Float,
+  modifier: Modifier = Modifier,
+  scale: BpkRatingScale? = BpkRatingScale.ZeroToFive,
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/rating/BpkRating.kt:65]: Showing the scale is different from the supported scale. For example the trip advisor example doesn't show the scale, but the logo still has a 0-5 scale, therefore we should clamp it to that. So the `showScale` property still makes sense and making this non-nullable.

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/rating/internal/BpkRatingComponents.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    style = BpkTheme.typography.bodyDefault,
+    color = BpkTheme.colors.textSecondary,
+    maxLines = 1,
+    overflow = TextOverflow.Ellipsis,
+  )
+}
+
+private fun formatValue(value: Float, scale: BpkRatingScale?, format: DecimalFormat): String {
+  val coerced = if (scale != null) value.coerceIn(scale.range) else value
+  return format.format(coerced)
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/rating/internal/BpkRatingComponents.kt:129]: how is this rounding? i remember the existing component had a specific rounding we had to use

Comment (bvitaliyg) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/rating/internal/BpkRatingComponents.kt:129]: https://github.com/Skyscanner/backpack-android/blob/936cfcb629c53317c407bcd626581bdab6c80c38/Backpack/src/main/java/net/skyscanner/backpack/rating/internal/RatingScore.kt#L38

This is what we use for rounding. Added it to the code

**Comment by bvitaliyg** [comment]
Record snapshots

**Pull request merged**